### PR TITLE
Extract FakeFileSystem into a new okio-testing module

### PR DESCRIPTION
**Pull request by swankjesse**
I want FakeFileSystem to depend on experimental APIs in kotlinx.datetime.
And I want to fold okio-files into the main okio module, where a dependency
on kotlinx.datetime would be too much.

### CODE REVIEW

**Review by Egorand — APPROVED** [review]

**Comment by yogurtearl** [comment]
@swankjesse any reason to not move `FakeFilesystemTest` over to `okio-testing` as well?

**Comment by swankjesse** [comment]
@yogurtearl it shares code with the real Filesystem test, and it's hard to make the dependencies line up.

File: okio-testing/build.gradle [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apply plugin: 'org.jetbrains.kotlin.multiplatform'
+
+kotlin {
+  jvm {
+    withJava()
```
Comment (JakeWharton) [inline okio-testing/build.gradle:5]: What Java code is in this module?

Comment (swankjesse) [reply okio-testing/build.gradle:5]: None. I’m attempting to configure this as close as possible to the main `okio` artifact so I can later merge the two without much friction.

**Pull request merged**